opta/upgrade.py

```diff
@@ -25,28 +27,37 @@ def _get_latest_version() -> str:
     return resp.text.strip().strip("v")
 
 
-def check_version_upgrade() -> None:
+def check_version_upgrade(is_upgrade_call: bool = False) -> bool:
     """Logs a warning if newer version of opta is available.
 
     The version check is not always performed when this function is called.
     It is performed non-deterministically with a probability of UPGRADE_CHECK_PROBABILITY
     in order to not spam the user.
     """
-    if not _should_check_for_version_upgrade():
-        return
-    logger.debug("Checking for version upgrades...")
-    try:
-        latest_version = _get_latest_version()
-    except Exception as e:
-        logger.debug(e, exc_info=True)
-        logger.debug("Unable to find latest version.")
-        return
-    try:
-        if semver.VersionInfo.parse(VERSION.strip("v")).compare(latest_version) < 0:
-            logger.warning(
-                "New version available.\n"
-                f"You have {VERSION} installed. Latest version is {latest_version}.\n"
-                f"Upgrade instructions are available at {UPGRADE_INSTRUCTIONS_URL}."
-            )
-    except Exception as e:
-        logger.debug(f"Semver check failed with error {e}")
+    if Path(successfull_upgrade).is_file():
+        os.remove(successfull_upgrade)
```

Comment:
I am not sure I follow this one, why is using a local file needed?

Reply:
I added this one so as to not execute the check_version functionality if the update happens successfully.
Created a local file rather that setting up the Environment variable, because the implementation of the `os.unsetenv` differs for Linux and OSX machines

Reply:
is the issue to make sure that `check_version_upgrade` is only called once?
using a local file to save some state can be seen as [a side effect](https://en.wikipedia.org/wiki/Side_effect_(computer_science)).
would something like [this](https://stackoverflow.com/a/29764451) work?

Reply:
Used [this](https://stackoverflow.com/a/29764379) to remove the Side Effect